File: src/bin/0338-counting-bits.py
# @leetup=info id=338 lang=python3 slug=counting-bits



# @leetup=code
class Solution:
    def countBits(self, n: int) -> list[int]:
        res = [0] * (n + 1)
        for i in range(n + 1):
            res[i] = res[i // 2] + (i % 2)
        return res


# @leetup=code

File: src/bin/0338-counting-bits.rs
// @leetup=info id=338 lang=rust slug=counting-bits

// @leetup=code
impl Solution {
    pub fn count_bits(n: i32) -> Vec<i32> {
        let mut res = vec![0; n as usize + 1];
        for i in 0..(n as usize + 1) {
            res[i] = res[i / 2] + (i as i32) % 2;
        }
        res
    }
}
// @leetup=code

struct Solution;

fn main() {
    let _ = Solution::count_bits;
}
